drv/lpc55-swd/src/main.rs

```diff
@@ -651,28 +639,32 @@ impl NotificationHandler for ServerImpl {
         //     and don't complicate the behavior here.
         //
 
-        let mut invalidate = false;
         let gpio = Pins::from(self.gpio);
 
         if (bits & notifications::JTAG_DETECT_IRQ_MASK) != 0 {
             ringbuf_entry!(Trace::SpJtagDetectFired);
             const SLOT: PintSlot = SP_TO_ROT_JTAG_DETECT_L_PINT_SLOT;
-            if let Ok(Some(detected)) =
-                gpio.pint_op(SLOT, PintOp::Detected, PintCondition::Falling)
-            {
+            if let Ok(detected) = gpio.pint_detect(SLOT, PintFlag::Falling) {
                 if detected {
                     ringbuf_entry!(Trace::InvalidateSpMeasurement);
                     // Reset the attestation log
-                    invalidate = true;
-                    self.next_use_must_setup_swd();
+                    self.invalidate_sp_measurement();
+
+                    // Cancel ongoing transactions
+                    self.transaction = None;
+                    self.state = SwdState::Disconnected;
+                    switch_io_disconnected();
                 }
+                let _ = gpio.pint_clear(SLOT, PintFlag::Both);
             } else {
-                // The pint_op parameters are for a configured PINT slot for one of
-                // our configured GPIO pins. We're testing a status bit in a register.
-                unreachable!();
+                // Spurious notification, ignore it
             }
-            let _ = gpio.pint_op(SLOT, PintOp::Clear, PintCondition::Status);
             sys_irq_control(notifications::JTAG_DETECT_IRQ_MASK, true);
+        } else if matches!(self.state, SwdState::Disconnected)
+            && !self.swd_dongle_detected()
+        {
+            switch_io_connected();
+            self.state = SwdState::Connected { init: false };
```

Comment:
So, reading over this change, I'm having a hard time convincing myself that every update to this state is accompanied by an actual reconfiguration of the pins. Should the maintenance of the state variable move into the various `switch_io_*` routines to simplify reasoning locally about it?